apps/backend/src/app/modules/killswitch/killswitch.service.ts

```diff
@@ -0,0 +1,106 @@
+import { ExtendedPrismaService } from '../database/prisma.extension';
+import { KillswitchType } from './killswitch.enum';
+import { Enum } from '@momentum/enum';
+
+export type Killswitches = Record<KillswitchType, boolean>;
+export const KILLSWITCH_CONFIG_ID = 'Killswitches';
+
+@Injectable()
+export class KillswitchService implements OnModuleInit {
+  constructor(
+    @Inject(EXTENDED_PRISMA_SERVICE) private readonly db: ExtendedPrismaService
+  ) {}
+
+  private switches: Killswitches;
+  private readonly logger = new Logger('killswitch service');
+
+  public async onModuleInit() {
+    const dbResponse = await this.db.config.findUnique({
+      where: { id: KILLSWITCH_CONFIG_ID }
+    });
+
+    const storedSwitches = dbResponse?.value as Killswitches;
+    if (storedSwitches) {
+      const areValidSwitches = this.validateKillswitches(storedSwitches);
+      if (areValidSwitches) {
+        this.switches = storedSwitches;
+        this.logger.log(
+          `Killswitches found, current killswitches are: ${JSON.stringify(storedSwitches)}`
+        );
+      } else {
+        this.logger.error('Stored killswitches failed validation.');
+        this.switches = {} as Killswitches;
+        await this.updateKillswitches(
+          Object.fromEntries(
+            Enum.values(KillswitchType).map((type: KillswitchType) => [
+              type,
+              false
+            ])
+          ) as Killswitches
+        );
+      }
+    } else {
+      this.logger.log(
+        'No killswitches found, creating default switches as disabled.'
+      );
+      this.switches = {} as Killswitches;
+      await this.updateKillswitches(
+        Object.fromEntries(
+          Enum.values(KillswitchType).map((type: KillswitchType) => [
+            type,
+            false
+          ])
+        ) as Killswitches
+      );
+    }
+  }
+
+  public checkKillswitch(type: KillswitchType): boolean {
+    return this.switches[type];
+  }
+
+  public async updateKillswitches(switches: Killswitches): Promise<void> {
+    const areValidSwitches = this.validateKillswitches(switches);
+
+    if (!areValidSwitches) {
+      throw new BadRequestException();
+    }
+
+    for (const [type, value] of Object.entries(switches)) {
+      this.switches[type] = value;
+    }
+
+    await this.db.config.upsert({
+      where: { id: KILLSWITCH_CONFIG_ID },
+      update: { value: this.switches },
+      create: { id: KILLSWITCH_CONFIG_ID, value: this.switches }
+    });
+  }
+
+  public async getKillSwitches() {
+    return await this.db.config.findUnique({
+      where: { id: KILLSWITCH_CONFIG_ID }
+    });
+  }
+
+  private validateKillswitches(switches: Killswitches): boolean {
+    const switchTypes = Enum.values(KillswitchType);
+    for (const [type, value] of Object.entries(switches)) {
+      if (
+        !(value === true || value === false) ||
+        !switchTypes.includes(type as KillswitchType)
+      ) {
+        return false;
+      }
+    }
+    return true;
```

Comment:
Oh just realised, this can just be 
```ts
return Object.entries(switches).every(([type, value]) => (value === true || value === false) && switchTypes.includes(type as KillswitchType))
```
(let prettier do the heavy lifting)